example/Makefile.inc

```diff
@@ -13,4 +13,8 @@ AM_CFLAGS = \
 	-I$(top_srcdir)/platform/@with_platform@/arch/@ARCH_DIR@ \
 	-I$(top_builddir)/include
 
+if STATIC_LINK
 AM_LDFLAGS = -L$(LIB) -static
+else
+AM_LDFLAGS =
```

Comment:
I am using something similar for ODP code instrumentation with PAPI project in Caterpillar. Yet, I am surprised you don't get link errors when building examples with shared libs.

+if CODE_INSTRUM
 +AM_LDFLAGS = -L$(LIB) -lssl -lcrypto -latomic
 +else
  AM_LDFLAGS = -L$(LIB) -static
 +endif

Reply:
Hmm. Strange. Those libs should come from -lodp-linux, because they are included into .la file.